cmd/kube-apiserver/app/server.go

```diff
@@ -583,7 +584,19 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 		storageFactory.SetEtcdLocation(groupResource, servers)
 	}
 
-	if s.Etcd.EncryptionProviderConfigFilepath != "" {
+	// Set up encryption at rest via transformer overrides if a configuration file was provided.
+	if len(s.Etcd.EncryptionProviderConfigFilepath) != 0 {
+		cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
```

Comment:
this is assuming you have a cloud provider, which is not always true. if you don't, InitCloudProvider can return `nil, nil`